pkg/descheduler/controllers/migration/filter.go

```diff
@@ -317,6 +333,20 @@ func (r *Reconciler) filterMaxMigratingOrUnavailablePerWorkload(pod *corev1.Pod)
 		}
 	}
 
+	podJobName := getPodMigratingJobName(pod)
+	podOwnJob := &sev1alpha1.PodMigrationJob{}
+	if podJobName != "" {
+		err := r.Client.Get(context.TODO(), client.ObjectKey{Name: podJobName}, podOwnJob)
+		if err != nil && !errors.IsNotFound(err) {
+			klog.Errorf("failed to get PodMigrationJob, err: %v", err)
+			return false
+		}
+	}
+
+	if evictionsutil.HaveEvictAnnotation(podOwnJob) {
```

Comment:
IMO it's better to skip filter if job has evict annotation to keep the same semantics with Pod.